### PR TITLE
fix: removeMessage not working due to outdated messages state

### PR DESCRIPTION
**Pull request by tanasebutcaru**
#### Description

The `removeMessage` from `useMessages()` and `params` doesn't work because the `messages` are not updated within the `useCallback` wrapper of this function.

Closes #(issue)

#### What change does this PR introduce?

- added `eslint-plugin-react-hooks` devDependecy
- updated .eslintrc config to use the react-hooks and display warnings related to react hooks
- with the above configuration, the editor (VSCode) started to provide suggestions related to missing react hooks dependencies
- added the missing dependencies for `removeMessage` function
- added missing dependencies for other useCallback functions from the `useMessagesInternal.ts` and `useSubmitInputInternal.ts` files
- moved the `simulateStream` function before `injectMessage` since it was used before it was defined

Please select the relevant option(s).

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (changes to docs/code comments)

#### What is the proposed approach?

Always add the required dependencies for a react hook.
The new eslint plugin for react hooks will help development by providing warnings when writing new code and when executing the lint command.  

#### Checklist:

- [x] The commit message follows our adopted [guidelines](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Testing has been done for the change(s) added (for bug fixes/features)
- [ ] Relevant comments/docs have been added/updated (for bug fixes/features)

### CODE REVIEW

File: src/hooks/internal/useMessagesInternal.ts [hunk 1]
```diff
@@ -31,6 +31,64 @@ export const useMessagesInternal = () => {
...
 
+	/**
+	 * Simulates the streaming of a message from the bot.
+	 * 
+	 * @param message message to stream
+	 * @param streamSpeed speed to stream the message
+	 * @param useMarkup boolean indicating whether markup is used
+	 */
+	const simulateStream = useCallback(async (message: Message, streamSpeed: number, useMarkup: boolean) => {
+	// stop bot typing when simulating stream
```
Comment (tjtanjin) [inline src/hooks/internal/useMessagesInternal.ts:42]: Comment is not aligned properly with the statement it belongs to, a few other instances of this below

**Review by tjtanjin — CHANGES_REQUESTED** [review]
Generally alright, just a few nitpicks:
- Align comments to their statements
- Remove `refs` and `state setters` from dependency arrays of `useCallback` (our current use case does not require `refs` and `state setters` as a dependency)

File: src/hooks/internal/useMessagesInternal.ts (hunk 1 above)
Comment (tanasebutcaru) [reply src/hooks/internal/useMessagesInternal.ts]: Pushed.

**Review by tjtanjin — APPROVED** [review]
LGTM!

**Pull request merged**